File: Python/392_Is_Subsequence.py
# Given two strings s and t, return true if s is a subsequence of t, or false otherwise.

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i, j = 0, 0
        while(i<len(s) and j<len(t)):
            if( s[i] == t[j]):
                i+=1
            j+=1
        
        return True if(i == len(s)) else False
        

File: C++/392_Is_Subsequence.cpp
#include<string>
using namespace std;
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0;
        int j = 0;
        
        while( i< s.size() and j<t.size())
        {
            if (s[i] == t[j])
            {
                i = i+1;
            }
            j = j+1;
        }
        
        if (i == s.size())
        {
            return true;
        }
        else
        {
            return false;
        }
    }
};